internal/models/users/user_ranking/user_ranking.go

```diff
@@ -16,21 +16,23 @@ type UserRanking struct {
 func GetRanking(db *sql.DB) ([]UserRanking, error) {
 	rows, err := db.Query(`
 SELECT 
-CONCAT(username_adjective, ' ', username_noun) AS username,
-total_points,
-placement
-
-FROM users
-JOIN (
-SELECT
-   user_id,
-   SUM(points_awarded) AS total_points,
-   RANK() OVER (ORDER BY SUM(points_awarded) DESC) AS placement
-FROM user_answers
-GROUP BY user_id
-) AS ua ON ua.user_id =users.id
-
-WHERE opt_in_ranking = true;
+    CONCAT(username_adjective, ' ', username_noun) AS username,
+    COALESCE(total_points, 0) AS total_points,
```

Comment:
huh didnt know `COALESCE` was a thing. You couldve also used [sql.NullInt32](https://pkg.go.dev/database/sql#NullInt32) or [sql.NullInt64](https://pkg.go.dev/database/sql#NullInt64) to handle null values in go, see https://go.dev/doc/database/querying#nullable_columns

but this works i htink so shall be fine